src/main/webapp/app/exercises/quiz/shared/questions/multiple-choice-question/multiple-choice-question.component.ts

```diff
@@ -78,9 +79,10 @@ export class MultipleChoiceQuestionComponent {
             return;
         }
         if (this.isAnswerOptionSelected(answerOption)) {
-            this.selectedAnswerOptions = this.selectedAnswerOptions.filter((selectedAnswerOption) => selectedAnswerOption.id !== answerOption.id);
+            const options = this.selectedAnswerOptions.get(this.questionIndex)?.filter((selectedAnswerOption) => selectedAnswerOption.id !== answerOption.id) || [];
+            this.selectedAnswerOptions.set(this.questionIndex, options);
         } else {
-            this.selectedAnswerOptions.push(answerOption);
+            this.selectedAnswerOptions.get(this.questionIndex)!.push(answerOption);
```

Comment:
So participating in a quiz seems to be broken. I'm not 100% sure why but i think it has something to do with this line and similar lines for other question types.
Here the selectedAnswerOption uses its get() function with the questionIndex as key, but in the quiz-participation component the keys are the question ids, not the indexes.

Reply:
The issue seems to affect multiple-choice and d&d questions, because of the change from array to map in line 29 (this component) / 51 (d&d component).

Reply:
This is now fixed